### PR TITLE
use random port for solr tests

### PR DESCRIPTION
**Pull request by mkalkbrenner**
as suggested by @Bo98 in #50759

### CODE REVIEW

File: Formula/solr.rb [hunk 1]
```diff
@@ -59,17 +59,23 @@ def plist
...
 
   test do
+    require "socket"
+
+    server = TCPServer.new(0)
+    port = server.addr[1]
+    server.close
+
     # Info detects no Solr node => exit code 3
     shell_output(bin/"solr -i", 3)
```
Comment (Bo98) [inline Formula/solr.rb]: This currently checks 8983. Is `-p` supported here too?

Comment (mkalkbrenner) [reply Formula/solr.rb]: no, `solr -i` uses the pid files to detect the ports and sends a request to each of them:
```
solr -i

Found 2 Solr nodes: 

Solr process 91692 running on port 43566
{
  "solr_home":"/usr/local/var/lib/solr",
  "version":"8.4.1 832bf13dd9187095831caf69783179d41059d013 - ishan - 2020-01-10 13:40:28",
  "startTime":"2020-02-28T09:43:07.334Z",
  "uptime":"0 days, 0 hours, 0 minutes, 6 seconds",
  "memory":"30.7 MB (%6) of 512 MB"}


Solr process 91104 running on port 8983
{
  "solr_home":"/usr/local/solr",
  "version":"8.4.1 832bf13dd9187095831caf69783179d41059d013 - ishan - 2020-01-10 13:40:28",
  "startTime":"2020-02-28T09:37:18.046Z",
  "uptime":"0 days, 0 hours, 5 minutes, 57 seconds",
  "memory":"1.2 GB (%29.6) of 4 GB",
  "cloud":{
    "ZooKeeper":"localhost:9983",
    "liveNodes":"1",
    "collections":"8"}}
```

Comment (mkalkbrenner) [reply Formula/solr.rb]:
```
/usr/local/var/run/solr] $ ls
solr-43566.pid solr-8983.pid
```

Comment (Bo98) [reply Formula/solr.rb]: Looks like it also does `ps auxww | grep start\.jar | grep solr\.solr\.home | grep -v grep | awk '{print $2}' | sort -r`. The tests were picking up a server that wasn't terminated properly but didn't have a PID file.

We'll cover that more generally - we usually kill processes at the end of every CI run but that isn't working on GitHub Actions at the moment.

**Pull request merged**